lib/core/src/server/config.js

```diff
@@ -0,0 +1,124 @@
+/* eslint-disable global-require, import/no-dynamic-require */
+import fs from 'fs';
+import path from 'path';
+import findCacheDir from 'find-cache-dir';
+import { logger } from '@storybook/node-logger';
+import { createDefaultWebpackConfig } from './config/defaults/webpack.config';
+import loadBabelConfig from './babel_config';
+
+const noopWrapper = config => config;
+
+function setBabelLoaderOptions(config, configDir, defaultBabelConfig, wrapBabelConfig) {
+  const jsRule = config.module.rules[0];
+
+  const babelConfig = wrapBabelConfig(loadBabelConfig(configDir, defaultBabelConfig));
+
+  jsRule.query = {
+    // This is a feature of `babel-loader` for webpack (not Babel itself).
+    // It enables a cache directory for faster-rebuilds
+    // `find-cache-dir` will create the cache directory under the node_modules directory.
+    cacheDirectory: findCacheDir({ name: 'react-storybook' }),
+    ...babelConfig,
+  };
+}
+
+function setConfigJsToEntry(config, configDir) {
+  // Check whether a config.js file exists inside the storybook
+  // config directory and throw an error if it's not.
+  const storybookConfigPath = path.resolve(configDir, 'config.js');
+  if (!fs.existsSync(storybookConfigPath)) {
+    throw new Error(`=> Create a storybook config file in "${configDir}/config.js".`);
+  }
+
+  config.entry.preview.push(require.resolve(storybookConfigPath));
+}
+
+function setAddonsJsToEntry(config, configDir) {
+  // Check whether addons.js file exists inside the storybook.
+  const storybookCustomAddonsPath = path.resolve(configDir, 'addons.js');
+  if (fs.existsSync(storybookCustomAddonsPath)) {
+    logger.info('=> Loading custom addons config.');
+    config.entry.manager.unshift(storybookCustomAddonsPath);
+  }
+}
+
+function mergeConfigs(config, customConfig) {
+  return {
+    ...customConfig,
+    // We'll always load our configurations after the custom config.
+    // So, we'll always load the stuff we need.
+    ...config,
+    // Override with custom devtool if provided
+    devtool: customConfig.devtool || config.devtool,
+    // We need to use our and custom plugins.
+    plugins: [...config.plugins, ...(customConfig.plugins || [])],
+    module: {
+      ...config.module,
+      // We need to use our and custom rules.
+      ...customConfig.module,
+      rules: [
+        ...config.module.rules,
+        ...((customConfig.module && customConfig.module.rules) || []),
+      ],
+    },
+    resolve: {
+      ...config.resolve,
+      ...customConfig.resolve,
+      alias: {
+        ...config.alias,
+        ...(customConfig.resolve && customConfig.resolve.alias),
+      },
+    },
+  };
+}
+
+function informAboutCustomConfig(defaultConfigName) {
+  if (!defaultConfigName) {
+    logger.info('=> Using default webpack setup.');
+    return;
+  }
+
+  logger.info(`=> Using default webpack setup based on "${defaultConfigName}".`);
```

Comment:
Actually, default config is always the same, and it's based on create-react-app. So when we say that it's based e.g. on polymer-cli, it's not true